### PR TITLE
Move MO to openvino package

### PR DESCRIPTION
**Pull request by popovaan**
Draft PR for split MO to lite and legacy and moving lite MO to openvino package.
Ticket: 
98999

### CODE REVIEW

File: src/bindings/python/src/openvino/tools/mo_lite/__main__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from openvino.tools.mo_lite.subprocess_main import subprocess_main
+from openvino.tools.mo_lite.utils.telemetry_utils import init_mo_telemetry
+init_mo_telemetry()
+subprocess_main(framework=None)
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/tools/mo_lite/__main__.py:7]: Is it required to be a subprocess?

File: src/bindings/python/src/openvino/tools/mo_lite/subprocess_main.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def check_python_version():
+    """
+    Checks python version to be greater or equal than 3.4
+    :return: exit code (1 - error, None - successful)
+    """
+    if sys.version_info < (3, 4):
+        print('Python version should be of version 3.4 or newer')
+        return 1
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/tools/mo_lite/subprocess_main.py:17]: 3.7?

File: src/bindings/python/src/openvino/tools/mo_lite/subprocess_main.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def log_ie_not_found():
+    log.error("Could not find the Inference Engine or nGraph Python API.\n"
+              "Consider building the Inference Engine and nGraph Python APIs"
+              " from sources or try to install OpenVINO (TM) Toolkit using pip \npip install openvino")
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/tools/mo_lite/subprocess_main.py:23]: Inference Engine or nGraph Python? Is API2.0 or 1.0 used?

File: src/bindings/python/src/openvino/tools/mo_lite/subprocess_main.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def log_mo_root_dir_not_found():
+    log.error("Could not find the ModelOptimizer root module directory.\n"
+              "Consider setting PYTHONPATH to the openvino tools folder (usually openvino/tools/mo)")
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/tools/mo_lite/subprocess_main.py:28]: `openvino/tools/mo` to mo-lite version

**Review by jiwaszki — COMMENTED** [review]
I know this is "do not review" but it is better to think about this on early stage

File: src/bindings/python/src/openvino/tools/mo_lite/subprocess_main.py (hunk 4 above)
Comment (popovaan) [reply src/bindings/python/src/openvino/tools/mo_lite/subprocess_main.py:28]: Most likely mo-lite will be renamed to mo and the rest of MO to mo_legacy. mo-lite is temporary name.

File: tools/mo/openvino/tools/mo/mo.py [hunk 5]
```diff
@@ -1,11 +1,4 @@
...
-
 # Copyright (C) 2018-2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-
-if __name__ == "__main__":
-    from openvino.tools.mo.utils.telemetry_utils import init_mo_telemetry
-    init_mo_telemetry()
-    from openvino.tools.mo.subprocess_main import subprocess_main  # nosec; pylint: disable=no-name-in-module
-    subprocess_main(framework=None)
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/mo.py:11]: @popovaan this change disables entry point. You can not import `if __name__ == "__main__":` statement from another file.

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Pull request closed without merging**